### PR TITLE
App refactorization

### LINKED ISSUE #76: Application logic refactorization
Based in PR #62 comments, a refactorization for the server is needed. The main goals are:
- Fix/remove the 'promise hell', and remove the nested code
- Refactor the code used in several places
- Remove the 'filtering functions' and start using sql (simple query language) from Orion
- Adapt requests to the new Orion data model

### PR DESCRIPTION
**Pull request by albertinisg**
This PR fixes #76 and is based on the feedback received in #62. It includes the following changes:
- Fix/remove the 'promise hell', and remove the nested code
- Refactor the code used in several places
- Remove the 'filtering functions' and start using sql (simple query language) from Orion
- Decrease the number of requests against Orion
- Adapt requests to the new Orion data model 
- Add comments explaining the functions at utils.js
- Minor lint issues
- New data image
- Fix travis build

@jmcanterafonseca could you have a look please? Thanks!

### CODE REVIEW

File: server/routes/orion.js [hunk 1]
```diff
@@ -32,231 +32,141 @@ tv4.addSchema('review', schema.review);
...
-          res.statusCode = err.statusCode;
-          res.json(err.error);
-        });
-    })
-    .catch(function(err) {
-      res.statusCode = err.statusCode;
-      res.end();
-    });
+  var restaurantId = utils.generateId(req.params.id);
+  var actualDate = new Date().toISOString();
```
Comment (jmcanterafonseca) [inline server/routes/orion.js:60]: nit: currentDate

File: server/routes/orion.js [hunk 2]
```diff
@@ -265,541 +175,390 @@ exports.getOrganizationRestaurants = function(req, res) {
...
+      utils.addConditionToQuery(sqlList, 'startTime', '==', timeframe);
+      utils.addConditionToQuery(sqlList, 'reservationFor', '==', req.params.id);
+      utils.addConditionToQuery(sqlList, 'reservationStatus', '==',
+                                'Confirmed');
+      queryString.q = sqlList.join(';');
+      return utils.getListByType('FoodEstablishmentReservation', null,
+                                 req.headers, queryString);
+    })
+    .then(function(reservations) {
+      actualOccupancyLevels = utils.getOccupancyLevels(reservations.body);
```
Comment (jmcanterafonseca) [inline server/routes/orion.js:403]: currentOccupancyLevels

**Comment by jmcanterafonseca** [comment]
thanks for the effort, could we split this work in 2 or 3 PRs, it would make easier the review.

for instance. refactoring and data model, would be a good approach. What do you think?

**Comment by albertinisg** [comment]
Sounds good, I'll work on it

**Comment by albertinisg** [comment]
Closing this (duplicated)

**Pull request closed without merging**